lib/migrations/update_accessible_objects_filter_performance.rb

```diff
@@ -0,0 +1,297 @@
+            filter_conditions := filter_conditions || filter_key || ' = ' || filter_value || ' AND ';
+          END LOOP;
+
+          /* Chomp trailing AND */
+          filter_conditions := left(filter_conditions, -4);
+          /* Replace double quotes */
+          filter_conditions := replace(filter_conditions, '"', '''');
+
+          EXECUTE format(
+            'INSERT INTO t_user_attribute_filtered_ids ' ||
+            'SELECT ua.child_id ' ||
+            'FROM t_user_attribute_ids ua ' ||
+            'JOIN policy_elements pe ' ||
+            '  ON pe.id = ua.child_id ' ||
+            'WHERE %s',
+            filter_conditions
+          );
+
+          user_attribute_table := 't_user_attribute_filtered_ids';
+        END IF;
+
+        EXECUTE format(
+          'INSERT INTO t_operation_set_ids ' ||
+          'SELECT ' ||
+          '  pea.operation_set_id, ' ||
+          '  pea.object_attribute_id ' ||
+          'FROM ' ||
+          '  policy_element_associations pea ' ||
+          '  JOIN %I t ON t.child_id = pea.user_attribute_id',
+          user_attribute_table
+        );
+
+        WITH RECURSIVE accessible_operations AS (
+          (
+            SELECT
+              child_id,
+              parent_id AS operation_set_id
+            FROM assignments
+            WHERE parent_id IN (SELECT operation_set_id FROM t_operation_set_ids)
+          )
+          UNION ALL
+          (
+            SELECT
+              a.child_id,
+              op.operation_set_id AS operation_set_id
+            FROM
+              assignments a
+              JOIN accessible_operations op ON op.child_id = a.parent_id
+          )
+        )
+        INSERT INTO t_accessible_operations
+        SELECT * FROM accessible_operations;
+
+        INSERT INTO t_operation_sets
+        SELECT DISTINCT ao.operation_set_id, ops.unique_identifier
+          FROM
+            t_accessible_operations ao
+            JOIN policy_elements ops ON ops.id = ao.child_id
+          WHERE ops.unique_identifier = ANY (operations);
+
+        RETURN QUERY EXECUTE
+        format(
+          'SELECT os.unique_identifier, array_agg(DISTINCT pe.%I) AS objects ' ||
+          'FROM ' ||
+          '  t_operation_set_ids os_id ' ||
+          '  JOIN t_operation_sets os ON os.operation_set_id = os_id.operation_set_id ' ||
+          '  JOIN policy_elements pe ON pe.id = os_id.object_attribute_id ' ||
+          'WHERE pe."type" = ''PolicyMachineStorageAdapter::ActiveRecord::Object'' ' ||
+          'GROUP BY os.unique_identifier',
+          field
+        );
+
+        TRUNCATE t_user_attribute_ids;
+        TRUNCATE t_user_attribute_filtered_ids;
+        TRUNCATE t_operation_set_ids;
+        TRUNCATE t_accessible_operations;
+        TRUNCATE t_operation_sets;
+
+        RETURN;
+      EXCEPTION
+        WHEN read_only_sql_transaction THEN
+          RETURN QUERY SELECT * FROM pm_accessible_objects_for_operations_cte(user_id, operations, field, filters);
```

Comment:
What is this doing?

Reply:
This exception is raised when attempting to create a temporary table from a physical read replica. It's a weak fallback if the app is placed in read only mode.

Reply:
Oh, interesting. How often is `pm_accessible_objects_for_operations_cte` used? Does it exist solely for RR fallback?